src/databaseaccessor/commentdb.cpp

```diff
@@ -0,0 +1,113 @@
+#include "commentdb.h"
+#include <QDebug>
+CommentDb::CommentDb(): Accessor()
+{
+}
+
+void CommentDb::addCommentsToDb(const QVector<Comment> &comments)
+{
+    for (auto &i : comments)
+    {
+        execQuery(addCommentQuery(i));
+    }
+    query.finish();
+}
+
+void CommentDb::deleteCommentFromDb(const int commentLine, const QString commentFile)
+{
+    execQuery(deleteCommentQuery(commentLine, commentFile));
+    query.finish();
+}
+
+void CommentDb::deleteCommentsFromDb(const QString commentFile)
+{
+    execQuery(deleteAllCommentsInFileQuery(commentFile));
+}
+
+QVector<Comment> CommentDb::getAllCommentsFromFile(const QString filename)
+{
+      execQuery(numberOfCommentInFileQuery(filename));
+      query.first();
+      int count_of_messages =query.value(0).toInt();
+      QVector<Comment> comments(count_of_messages);
+      execQuery(allCommentInFileQuery(filename));
+      int counter = 0;
+      while (query.next()) {
```

Comment:
while (query.next()) 
**{**

Reply:
2) You go through all database records, but you already know the total count of records, why do you calculate it again? You can rewrite this cycle as:
`     for( auto i = 1; i < count_of_messages; ++i)
      {
          fillStructComment(comments[i]);
 }`